syntax = "proto3";

package proto;

message ReadRequest {
    string uuid = 1;
}

message ReadResponse {
    string status = 1;
    string name = 2;
    string content = 3;
    string timestamp = 4;
}

message WriteRequest {
    string name = 1;
    string content = 2;
    string uuid = 3;
}
 
message WriteResponse {
    string status = 1;
    string uuid = 2;
    string timestamp = 3;
}

message serverData {
    string name = 1;
    string port = 2;
}

message serverDataResponse {
    string status = 1;
}

message DeleteRequestServer{
    string uuid = 1;
    string timestamp = 2;
}

message DeleteRequest {
    string uuid = 1;
}

message DeleteResponse {
    string status = 1;
}

message WriteRequestServer { 
    string name = 1;
    string content = 2;
    string uuid = 3;
    string timestamp = 4;
}

service Server {
    rpc read(ReadRequest) returns (ReadResponse) {};
    rpc writeClientRequest(WriteRequest) returns (WriteResponse) {};
    rpc writeServerRequest(WriteRequestServer) returns (WriteResponse) {};
    rpc deleteServerRequest(DeleteRequestServer) returns (DeleteResponse) {};
    rpc deleteClientRequest(DeleteRequest) returns (DeleteResponse) {};
    rpc addNonPrimaryServers(serverData) returns (serverDataResponse) {};
}